app/assets/javascripts/discourse/app/static/prosemirror/core/plugin.js

```diff
@@ -0,0 +1,56 @@
+import { Plugin } from "prosemirror-state";
+
+export function extractNodeViews(extensions) {
+  return extensions.reduce((acc, { nodeViews }) => {
+    if (nodeViews) {
+      Object.entries(nodeViews).forEach(([name, NodeViewClass]) => {
+        acc[name] = (node, view, getPos) =>
+          new NodeViewClass(node, view, getPos);
+      });
+    }
+
+    return acc;
+  }, {});
```

Comment:
Maybe something like
```suggestion
  const allNodeViews = {};
  for (const { nodeViews } of extensions){
    if (nodeViews) {
      for(const [name, NodeViewClass] of Object.entries(nodeViews)){
        allNodeViews[name] = (node, view, getPos) => new NodeViewClass(node, view, getPos);
      }
    }
  }
  return allNodeViews;
```